client/post-editor/controller.js

```diff
@@ -138,6 +138,38 @@ const getAnalyticsPathAndTitle = ( postType, postId, postToCopyId ) => {
 	}
 };
 
+function waitForSiteIdAndSelectedEditor( context ) {
+	return new Promise( resolve => {
+		const unsubscribe = context.store.subscribe( () => {
+			const state = context.store.getState();
+			const siteId = getSelectedSiteId( state );
+			if ( ! siteId ) {
+				return;
+			}
+			const selectedEditor = getSelectedEditor( state, siteId );
+			if ( ! selectedEditor ) {
+				return;
+			}
+			unsubscribe();
+			resolve();
+		} );
+	} );
+}
+
+async function maybeCalypsoifyGutenberg( context, next ) {
+	await waitForSiteIdAndSelectedEditor( context );
```

Comment:
Almost, we'll get stuck when we already have siteId and editor preference in global state. Let's add a check to see if we need to `await` for these values.

![stuck](https://user-images.githubusercontent.com/1270189/48088581-405e8080-e1b7-11e8-9be3-b383ffa7b7f6.gif)